integration/test_collection_batch.py

```diff
@@ -39,33 +41,35 @@ def numpy(self) -> "MockNumpyTorch":
 
 
 @pytest.fixture(scope="function")
-def client_sync_indexing() -> Generator[weaviate.WeaviateClient, None, None]:
+def collection() -> Generator[weaviate.Collection, None, None]:
     client = weaviate.connect_to_local()
     client.collections.delete_all()
     client.collections.create(
-        name="Test",
+        name="CollectionBatchTest",
         properties=[
             Property(name="name", data_type=DataType.TEXT),
             Property(name="age", data_type=DataType.INT),
         ],
-        references=[ReferenceProperty(name="test", target_collection="Test")],
+        references=[ReferenceProperty(name="test", target_collection="CollectionBatchTest")],
     )
-    yield client
+    yield client.collections.get("CollectionBatchTest")
```

Comment:
now all tests have the same and could delete each others data if they happen to run in parallel, you could use a factory instead: https://github.com/weaviate/weaviate-python-client/pull/517#discussion_r1325686048

Reply:
That's true, but there's no deleting logic within the `client.batch` or `collection.batch` so such a thing shouldn't occur. If it does occur, it should go in a separate testing suite imo

Reply:
but the next line deletes the data - let's just leave it. I'll see if I can spend tomorrow in cleaning up the tests a bit